src/main/java/org/testeditor/fixture/host/s3270/Status.java

```diff
@@ -0,0 +1,355 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The returned status Information from s3270/ws3270 application after execution
+ * of a s3270 command. It consists of 2 lines. The first line represents the
+ * status message and the second line a return code like ok or error. The status
+ * message consists of 12 blank-separated fields. Here is an example for a
+ * complete status:
+ * <ul>
+ * <li>U U U C(abcdefg.google.mainfraim.com) I 2 24 80 11 45 0x0 -</li>
+ * <li>ok</li>
+ *
+ * The documentation about the fields can be found here: See <a href=
+ * "http://x3270.bgp.nu/x3270-script.html#Status-Format">http://x3270.bgp.nu/x3270-script.html#Status-Format</a>
+ */
+public class Status {
+
+    private static final Logger logger = LoggerFactory.getLogger(Status.class);
+
+    private KeyboardState keyboardState;
+    private ScreenFormatting screenFormatting;
+    private FieldProtection fieldProtection;
+    private ConnectionState connectionState;
+    private EmulatorMode emulatorMode;
+    private TerminalMode mode;
+    private int numberRows;
+    private int numberColumns;
+    private int currentCursorRow;
+    private int currentCursorColumn;
+    private String windowId;
+    private String executionTimeString;
+
+    public Status(String status) {
+        createStatus(status);
+    }
+
+    private void createStatus(String status) {
+        String[] splittedStatus = status.split(" ");
+        createKeyBoardState(splittedStatus[0]);
+        createScreenFormatting(splittedStatus[1]);
+        createFieldProtection(splittedStatus[2]);
+        createConnectionState(splittedStatus[3]);
+        createEmulatorMode(splittedStatus[4]);
+        createModelNumber(splittedStatus[5]);
+        createNumberOfRows(splittedStatus[6]);
+        createNumberOfColumns(splittedStatus[7]);
+        createCursorRow(splittedStatus[8]);
+        createCursorColumn(splittedStatus[9]);
+        createWindowId(splittedStatus[10]);
+        createCommandExecutionTime(splittedStatus[11]);
+    }
+
+    private void createKeyBoardState(String input) {
+        try {
+            setKeyboardState(KeyboardState.getKeyboardState(input));
+        } catch (StatusNotFoundException e) {
+            logger.error(e.getMessage());
+        }
+    }
+
+    private void createScreenFormatting(String input) {
+        try {
+            setScreenFormatting(ScreenFormatting.getScreenFormatting(input));
+        } catch (StatusNotFoundException e) {
+            logger.error(e.getMessage());
+        }
+    }
+
+    private void createFieldProtection(String input) {
+        try {
+            setFieldProtection(FieldProtection.getFieldProtection(input));
+        } catch (StatusNotFoundException e) {
+            logger.error(e.getMessage());
+        }
+    }
+
+    private void createConnectionState(String input) {
+        try {
+            String a = input.substring(0, 1);
```

Comment:
why only use the first character, isn't it an error if input is longer than one character?